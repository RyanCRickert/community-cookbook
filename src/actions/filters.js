export const setNameFilter = (name = "") => ({
    type: "SET_NAME_FILTER",
    name
});

export const sortByName = () => ({
    type: "SORT_BY_NAME",
});

export const sortByCategory = () => ({
    type: "SORT_BY_CATEGORY",
});

export const sortByCookTime = () => ({
    type: "SORT_BY_COOKTIME",
});

export const sortByFeeds = () => ({
    type: "SORT_BY_FEEDS",
});