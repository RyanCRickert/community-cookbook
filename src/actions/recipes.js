import uuid from "uuid";
import database from "../firebase/firebase";

export const addRecipe = (recipe) => ({
  type: "ADD_RECIPE",
  recipe
});

export const startAddRecipe = (recipeData = {}) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    const {
      name = "",
      ingredients = [],
      instructions = "",
      cookTime = 0,
      createdAt = 0
  } = recipeData;
    const recipe = { name, instructions, createdAt, ingredients, cookTime };

    return database.ref(`recipes`).push(recipe).then((ref) => {
      dispatch(addRecipe({
        id: ref.key,
        ...recipe
      }));
    });
  };
};

export const addIngredient = (id, ingredient) => ({
  type: "ADD_INGREDIENT",
  id,
  ingredient
});

export const startAddIngredient = (id, ingredient) => {
  return (dispatch) => {
    const {
      name = ""
    } = ingredientData;
    const ingredient = {name};

    return database.ref(`recipes/${id}/ingredients`).push(ingredient).then((ref) => {
      dispatch(addIngredient({
        iid: ref.key,
        ingredient
      }));
    });
  };
};

export const editIngredient = (id, iid, update) => {
  type:"EDIT_INGREDIENT",
  id,
  iid,
  update
};

export const startEditIngredient = (id, iid, update) => {
  return (dispatch) => {
    return database.ref(`recipes/${id}/ingredients${iid}`)
  };
};

export const removeRecipe = ({ id } = {}) => ({
  type: "REMOVE_RECIPE",
  id
});

export const startRemoveRecipe = ({ id }) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;

    return database.ref(`recipes/${id}`).remove().then(() => {
      dispatch(removeRecipe({ id }));
    });
  };
};

export const editRecipe = (id, updates) => ({
  type: "EDIT_RECIPE",
  id,
  updates
});

export const startEditRecipe = (id, updates) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;

    return database.ref(`recipes/${id}`).update(updates).then(() => {
      dispatch(editRecipe(id, updates));
    });
  };
};

export const setRecipes = (recipes) => ({
  type: "SET_RECIPES",
  recipes
});

export const startSetRecipes = () => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid

    return database.ref(`recipes`).once("value", (snapshot) => {
      const recipes = [];

      snapshot.forEach((childSnapshot) => {
        recipes.push({
          id: childSnapshot.key,
          ...childSnapshot.val()
        });
      });
      dispatch(setRecipes(recipes));
    });
  };
};

export const setIngredients = (id, ingredients) => ({
  type: "SET_INGREDIENTS",
  id,
  ingredients
});

export const startSetIngredients = () => {
  return (dispatch) => {
    return database.ref(`recipes/${id}/ingredients`).once("value", (snapshot) => {
      const ingredients = [];

      snapshot.forEach((childSnapshot) => {
        recipes.push({
          id: childSnapshot.key,
          ...childSnapshot.val()
        });
      });
      dispatch(setIngredients(ingredients));
    });
  };
};