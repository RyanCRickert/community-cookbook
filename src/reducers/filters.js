import moment from "moment";

const filtersReducerDefaultState = {
  name: "",
  sortBy: "name",
  category: ""
}

export default (state = filtersReducerDefaultState, action) => {
  switch (action.type) {
    case "SET_NAME_FILTER":
      return {
        ...state,
        name: action.name
      };
    case "SET_CATEGORY_FILTER":
      return {
        ...state,
        category: action.category
      };
    case "SORT_BY_NAME":
      return {
        ...state,
        sortBy: "name"
      };
    case "SORT_BY_COOKTIME":
      return {
        ...state,
        sortBy: "cookTime"
      }
    case "SORT_BY_FEEDS":
      return {
        ...state,
        sortBy: "feeds"
      }
    default:
      return state;
  }
};