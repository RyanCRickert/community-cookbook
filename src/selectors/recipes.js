import moment from "moment";

export default (recipes, {name, sortBy, startDate, endDate}) => {
    return recipes.filter((recipe) => {
        const createdAtMoment = moment(recipe.createdAt);
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, "day") : true ;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, "day") : true ;
        const nameMatch = recipe.name.toLowerCase().includes(name.toLowerCase());

        return startDateMatch && endDateMatch && nameMatch;
    }).sort((a , b) => {
        if (sortBy === "date") {
            return a.createdAt < b.createdAt ? 1 : -1;
        } else if (sortBy === "name") {
            return a.name > b.name ? 1 : -1;
        } else if (sortBy === "cookTime") {
            return a.cookTime < b.cookTime ? 1 : -1;
        } else if (sortBy === "feeds") {
            return a.feeds < b.feeds ? 1 : -1;
        }
    });
};