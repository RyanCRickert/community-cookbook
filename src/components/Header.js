import React from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import FontAwesome from "react-fontawesome";
import { startLogout } from "../actions/auth";
import { firebase } from "../firebase/firebase";
import LoginModal from "./LoginModal";

export class Header extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      loggedIn: null,
      modalOpen: false
    }

    this.handleOpenModal = this.handleOpenModal.bind(this);
    this.handleCloseModal = this.handleCloseModal.bind(this);
  }

  componentDidMount() {
    firebase.auth().onAuthStateChanged((user) => {
      if (user) {
        this.setState(() => ({ loggedIn: true }));
      } else {
        this.setState(() => ({ loggedIn: false }));
      }
    })
  }

  handleOpenModal() {
    this.setState(() => ({ modalOpen: true }));
  }

  handleCloseModal() {
    this.setState(() => ({ modalOpen: false }));
  }

  render() {
    return (
      <header className="header">
        <div className="content-container header-list">
          <div className="header-main">
            <h1 className="header-title" onClick={this.props.openWelcomeModal}>Community Cookbook</h1>
            <h3 className="header-subtitle">A place for sharing</h3>
          </div>
            {this.state.loggedIn
              ?
              <div className="header-buttons">
                <div className="header-log">
                  <Link className="button header-button" to="/add">New Recipe</Link>
                </div>
                <div className="header-log__fa">
                  <button className="button header-button" onClick={this.props.startLogout}>Logout</button>
                  <Link to="/cart"><div className="header-fa"><FontAwesome name="shopping-cart"/></div></Link>
                </div>
              </div>
              :
              <div className="header-log">
                <LoginModal modalOpen={this.state.modalOpen} handleCloseModal={this.handleCloseModal}/>
                <button className="button header-button header-button__login" onClick={this.handleOpenModal}>Login</button>
              </div>
            }
        </div>
      </header>
    )
  }

};

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header)