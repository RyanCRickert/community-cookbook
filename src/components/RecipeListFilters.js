import React from "react";
import { connect } from "react-redux";
import { setNameFilter, sortByName, sortByCookTime, sortByDate, sortByFeeds } from "../actions/filters";

export class RecipeListFilters extends React.Component {
  onNameChange = (e) => {
    this.props.setNameFilter(e.target.value);
  };

  onSortChange = (e) => {
    if (e.target.value === "date") {
      this.props.sortByDate();
    } else if (e.target.value === "cookTime") {
      this.props.sortByCookTime();
    } else if (e.target.value === "name") {
      this.props.sortByName();
    } else if (e.target.value === "feeds") {
      this.props.sortByFeeds();
    }
  };

  render() {
    return (
      <div className="content-container">
        <div className="input-group">
          <div className="input-group__item">
            <input
              type="text"
              className="text-input"
              placeholder="Search recipes"
              value={this.props.filters.name}
              onChange={this.onNameChange}
            />
          </div>
          <div className="input-group__item">
            <select
              className="select"
              value={this.props.filters.sortBy}
              onChange={this.onSortChange}
            >
              <option value="cookTime">Cook Time</option>
              <option value="date">Date</option>
              <option value="name">Name</option>
              <option value="feeds">Feeds</option>
            </select>
          </div>
        </div>
      </div>
    )
  }
};

const mapStateToProps = (state) => ({
  filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
  setNameFilter: (name) => dispatch(setNameFilter(name)),
  sortByDate: () => dispatch(sortByDate()),
  sortByName: () => dispatch(sortByName()),
  sortByCookTime: () => dispatch(sortByCookTime()),
  sortByFeeds: () => dispatch(sortByFeeds())
});

export default connect(mapStateToProps, mapDispatchToProps)(RecipeListFilters);