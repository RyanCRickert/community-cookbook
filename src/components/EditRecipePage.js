import React from "react";
import { connect } from "react-redux";
import RecipeForm from "./RecipeForm";
import { startEditRecipe, startRemoveRecipe } from "../actions/recipes";
import RemoveRecipeModal from "./RemoveRecipeModal";

export class EditRecipePage extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      modalOpen: false
    }

    this.handleOpenModal = this.handleOpenModal.bind(this);
    this.handleCloseModal = this.handleCloseModal.bind(this);
  }
  onSubmit = (recipe) => {
    this.props.startEditRecipe(this.props.recipe.id, recipe);
    this.props.history.push("/");
  };

  onRemove = () => {
    this.props.startRemoveRecipe({id : this.props.recipe.id});
    this.props.history.push("/");
  }

  handleOpenModal() {
    this.setState(() => ({ modalOpen: true }));
  }

  handleCloseModal() {
    this.setState(() => ({ modalOpen: false }));
  }
    
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Recipe</h1>
          </div>
        </div>
          <div className="content-container">
            <RecipeForm
              actionType={"Edit"}
              recipe={this.props.recipe}
              onSubmit={this.onSubmit}
            />
          <button className="button button--secondary" onClick={this.handleOpenModal}>Remove Recipe</button>
          <RemoveRecipeModal
          modalOpen={this.state.modalOpen}
          handleCloseModal={this.handleCloseModal}
          onRemove={this.onRemove}
          recipeName={this.props.recipe.name}
          />
          </div>
        </div>
    )}
}

const mapStateToProps = (state, props) => ({
  recipe: state.recipes.find((recipe) => recipe.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  startEditRecipe: (id, recipe) => dispatch(startEditRecipe(id, recipe)),
  startRemoveRecipe: (data) => dispatch(startRemoveRecipe(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditRecipePage);