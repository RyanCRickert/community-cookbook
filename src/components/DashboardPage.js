import React from "react";
import Header from "./Header";
import RecipeList from "./RecipeList";
import RecipeListFilters from "./RecipeListFilters";
import { Link } from "react-router-dom";

export default () => (
  <div className="content-container">
    <Header />
    <RecipeListFilters />
    <RecipeList />
  </div>
);