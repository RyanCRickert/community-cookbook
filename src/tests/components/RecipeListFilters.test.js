import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import { RecipeListFilters } from "../../components/RecipeListFilters";
import recipes from "../fixtures/recipes";
import { filters, altFilters } from "../fixtures/filters";

let setNameFilter, setCategoryFilter, sortByName, sortByCookTime, sortByFeeds, wrapper;

beforeEach(() => {
  setNameFilter = jest.fn();
  setCategoryFilter = jest.fn();
  sortByName = jest.fn();
  sortByCookTime = jest.fn();
  sortByFeeds = jest.fn();
  wrapper =  shallow(
    <RecipeListFilters
      filters={filters}
      setNameFilter={setNameFilter}
      setCategoryFilter={setCategoryFilter}
      sortByName={sortByName}
      sortByCookTime={sortByCookTime}
      sortByFeeds={sortByFeeds}
      />)
});

it("should render RecipeListFilters correctly", () => {
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("should render RecipeListFilters with alt data correctly", () => {
  wrapper.setProps({ filters: altFilters });
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("should handle name change", () => {
  const value = "some new text";

  expect(toJSON(wrapper)).toMatchSnapshot();
  wrapper.find("input").simulate("change", {
    target: { value }
  });
  expect(setNameFilter).toHaveBeenLastCalledWith(value);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("should handle sort by name", () => {
  const value = "name";

  expect(toJSON(wrapper)).toMatchSnapshot();
  wrapper.setProps({
    filters: altFilters
  });
  wrapper.find("select").at(1).simulate("change", {
    target: { value }
  });
  expect(sortByName).toHaveBeenCalled();
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("should handle sort by cook time", () => {
  const value = "cookTime";

  expect(toJSON(wrapper)).toMatchSnapshot();
  wrapper.find("select").at(1).simulate("change", {
    target: { value }
  });
  expect(sortByCookTime).toHaveBeenCalled();
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("should handle sort by feeds", () => {
  const value = "feeds";

  expect(toJSON(wrapper)).toMatchSnapshot();
  wrapper.find("select").at(1).simulate("change", {
    target: { value }
  });
  expect(sortByFeeds).toHaveBeenCalled();
  expect(toJSON(wrapper)).toMatchSnapshot();
});