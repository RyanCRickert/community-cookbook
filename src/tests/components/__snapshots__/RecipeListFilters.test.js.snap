// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle name change 1`] = `
<div
  className="content-container"
>
  <div
    className="input-group"
  >
    <div
      className="input-group__search"
    >
      <input
        className="text-input"
        onChange={[Function]}
        placeholder="Search recipes"
        type="text"
        value=""
      />
    </div>
    <div
      className="input-group__category"
    >
      <select
        className="category"
        onChange={[Function]}
      >
        <option
          value=""
        >
          All
        </option>
        <option>
          Beef
        </option>
        <option>
          Chicken
        </option>
        <option>
          Meatless
        </option>
        <option>
          Other
        </option>
        <option>
          Pork
        </option>
      </select>
    </div>
    <div
      className="input-group__select"
    >
      <select
        className="select"
        onChange={[Function]}
        value="date"
      >
        <option
          value="cookTime"
        >
          Cook Time
        </option>
        <option
          value="name"
        >
          Name
        </option>
        <option
          value="feeds"
        >
          Feeds
        </option>
      </select>
    </div>
  </div>
</div>
`;

exports[`should handle name change 2`] = `
<div
  className="content-container"
>
  <div
    className="input-group"
  >
    <div
      className="input-group__search"
    >
      <input
        className="text-input"
        onChange={[Function]}
        placeholder="Search recipes"
        type="text"
        value=""
      />
    </div>
    <div
      className="input-group__category"
    >
      <select
        className="category"
        onChange={[Function]}
      >
        <option
          value=""
        >
          All
        </option>
        <option>
          Beef
        </option>
        <option>
          Chicken
        </option>
        <option>
          Meatless
        </option>
        <option>
          Other
        </option>
        <option>
          Pork
        </option>
      </select>
    </div>
    <div
      className="input-group__select"
    >
      <select
        className="select"
        onChange={[Function]}
        value="date"
      >
        <option
          value="cookTime"
        >
          Cook Time
        </option>
        <option
          value="name"
        >
          Name
        </option>
        <option
          value="feeds"
        >
          Feeds
        </option>
      </select>
    </div>
  </div>
</div>
`;

exports[`should handle sort by cook time 1`] = `
<div
  className="content-container"
>
  <div
    className="input-group"
  >
    <div
      className="input-group__search"
    >
      <input
        className="text-input"
        onChange={[Function]}
        placeholder="Search recipes"
        type="text"
        value=""
      />
    </div>
    <div
      className="input-group__category"
    >
      <select
        className="category"
        onChange={[Function]}
      >
        <option
          value=""
        >
          All
        </option>
        <option>
          Beef
        </option>
        <option>
          Chicken
        </option>
        <option>
          Meatless
        </option>
        <option>
          Other
        </option>
        <option>
          Pork
        </option>
      </select>
    </div>
    <div
      className="input-group__select"
    >
      <select
        className="select"
        onChange={[Function]}
        value="date"
      >
        <option
          value="cookTime"
        >
          Cook Time
        </option>
        <option
          value="name"
        >
          Name
        </option>
        <option
          value="feeds"
        >
          Feeds
        </option>
      </select>
    </div>
  </div>
</div>
`;

exports[`should handle sort by cook time 2`] = `
<div
  className="content-container"
>
  <div
    className="input-group"
  >
    <div
      className="input-group__search"
    >
      <input
        className="text-input"
        onChange={[Function]}
        placeholder="Search recipes"
        type="text"
        value=""
      />
    </div>
    <div
      className="input-group__category"
    >
      <select
        className="category"
        onChange={[Function]}
      >
        <option
          value=""
        >
          All
        </option>
        <option>
          Beef
        </option>
        <option>
          Chicken
        </option>
        <option>
          Meatless
        </option>
        <option>
          Other
        </option>
        <option>
          Pork
        </option>
      </select>
    </div>
    <div
      className="input-group__select"
    >
      <select
        className="select"
        onChange={[Function]}
        value="date"
      >
        <option
          value="cookTime"
        >
          Cook Time
        </option>
        <option
          value="name"
        >
          Name
        </option>
        <option
          value="feeds"
        >
          Feeds
        </option>
      </select>
    </div>
  </div>
</div>
`;

exports[`should handle sort by feeds 1`] = `
<div
  className="content-container"
>
  <div
    className="input-group"
  >
    <div
      className="input-group__search"
    >
      <input
        className="text-input"
        onChange={[Function]}
        placeholder="Search recipes"
        type="text"
        value=""
      />
    </div>
    <div
      className="input-group__category"
    >
      <select
        className="category"
        onChange={[Function]}
      >
        <option
          value=""
        >
          All
        </option>
        <option>
          Beef
        </option>
        <option>
          Chicken
        </option>
        <option>
          Meatless
        </option>
        <option>
          Other
        </option>
        <option>
          Pork
        </option>
      </select>
    </div>
    <div
      className="input-group__select"
    >
      <select
        className="select"
        onChange={[Function]}
        value="date"
      >
        <option
          value="cookTime"
        >
          Cook Time
        </option>
        <option
          value="name"
        >
          Name
        </option>
        <option
          value="feeds"
        >
          Feeds
        </option>
      </select>
    </div>
  </div>
</div>
`;

exports[`should handle sort by feeds 2`] = `
<div
  className="content-container"
>
  <div
    className="input-group"
  >
    <div
      className="input-group__search"
    >
      <input
        className="text-input"
        onChange={[Function]}
        placeholder="Search recipes"
        type="text"
        value=""
      />
    </div>
    <div
      className="input-group__category"
    >
      <select
        className="category"
        onChange={[Function]}
      >
        <option
          value=""
        >
          All
        </option>
        <option>
          Beef
        </option>
        <option>
          Chicken
        </option>
        <option>
          Meatless
        </option>
        <option>
          Other
        </option>
        <option>
          Pork
        </option>
      </select>
    </div>
    <div
      className="input-group__select"
    >
      <select
        className="select"
        onChange={[Function]}
        value="date"
      >
        <option
          value="cookTime"
        >
          Cook Time
        </option>
        <option
          value="name"
        >
          Name
        </option>
        <option
          value="feeds"
        >
          Feeds
        </option>
      </select>
    </div>
  </div>
</div>
`;

exports[`should handle sort by name 1`] = `
<div
  className="content-container"
>
  <div
    className="input-group"
  >
    <div
      className="input-group__search"
    >
      <input
        className="text-input"
        onChange={[Function]}
        placeholder="Search recipes"
        type="text"
        value=""
      />
    </div>
    <div
      className="input-group__category"
    >
      <select
        className="category"
        onChange={[Function]}
      >
        <option
          value=""
        >
          All
        </option>
        <option>
          Beef
        </option>
        <option>
          Chicken
        </option>
        <option>
          Meatless
        </option>
        <option>
          Other
        </option>
        <option>
          Pork
        </option>
      </select>
    </div>
    <div
      className="input-group__select"
    >
      <select
        className="select"
        onChange={[Function]}
        value="date"
      >
        <option
          value="cookTime"
        >
          Cook Time
        </option>
        <option
          value="name"
        >
          Name
        </option>
        <option
          value="feeds"
        >
          Feeds
        </option>
      </select>
    </div>
  </div>
</div>
`;

exports[`should handle sort by name 2`] = `
<div
  className="content-container"
>
  <div
    className="input-group"
  >
    <div
      className="input-group__search"
    >
      <input
        className="text-input"
        onChange={[Function]}
        placeholder="Search recipes"
        type="text"
        value="chicken"
      />
    </div>
    <div
      className="input-group__category"
    >
      <select
        className="category"
        onChange={[Function]}
      >
        <option
          value=""
        >
          All
        </option>
        <option>
          Beef
        </option>
        <option>
          Chicken
        </option>
        <option>
          Meatless
        </option>
        <option>
          Other
        </option>
        <option>
          Pork
        </option>
      </select>
    </div>
    <div
      className="input-group__select"
    >
      <select
        className="select"
        onChange={[Function]}
        value="feeds"
      >
        <option
          value="cookTime"
        >
          Cook Time
        </option>
        <option
          value="name"
        >
          Name
        </option>
        <option
          value="feeds"
        >
          Feeds
        </option>
      </select>
    </div>
  </div>
</div>
`;

exports[`should render RecipeListFilters correctly 1`] = `
<div
  className="content-container"
>
  <div
    className="input-group"
  >
    <div
      className="input-group__search"
    >
      <input
        className="text-input"
        onChange={[Function]}
        placeholder="Search recipes"
        type="text"
        value=""
      />
    </div>
    <div
      className="input-group__category"
    >
      <select
        className="category"
        onChange={[Function]}
      >
        <option
          value=""
        >
          All
        </option>
        <option>
          Beef
        </option>
        <option>
          Chicken
        </option>
        <option>
          Meatless
        </option>
        <option>
          Other
        </option>
        <option>
          Pork
        </option>
      </select>
    </div>
    <div
      className="input-group__select"
    >
      <select
        className="select"
        onChange={[Function]}
        value="date"
      >
        <option
          value="cookTime"
        >
          Cook Time
        </option>
        <option
          value="name"
        >
          Name
        </option>
        <option
          value="feeds"
        >
          Feeds
        </option>
      </select>
    </div>
  </div>
</div>
`;

exports[`should render RecipeListFilters with alt data correctly 1`] = `
<div
  className="content-container"
>
  <div
    className="input-group"
  >
    <div
      className="input-group__search"
    >
      <input
        className="text-input"
        onChange={[Function]}
        placeholder="Search recipes"
        type="text"
        value="chicken"
      />
    </div>
    <div
      className="input-group__category"
    >
      <select
        className="category"
        onChange={[Function]}
      >
        <option
          value=""
        >
          All
        </option>
        <option>
          Beef
        </option>
        <option>
          Chicken
        </option>
        <option>
          Meatless
        </option>
        <option>
          Other
        </option>
        <option>
          Pork
        </option>
      </select>
    </div>
    <div
      className="input-group__select"
    >
      <select
        className="select"
        onChange={[Function]}
        value="feeds"
      >
        <option
          value="cookTime"
        >
          Cook Time
        </option>
        <option
          value="name"
        >
          Name
        </option>
        <option
          value="feeds"
        >
          Feeds
        </option>
      </select>
    </div>
  </div>
</div>
`;
